pkg/ccl/streamingccl/streamingest/stream_ingestion_dist.go

```diff
@@ -185,12 +201,26 @@ func startDistIngestion(
 type replicationFlowPlanner struct {
 	// initial contains the stream addresses found during the replicationFlowPlanner's first makePlan call.
 	initialStreamAddresses []string
+
+	srcTenantID roachpb.TenantID
 }
 
 func (p *replicationFlowPlanner) containsInitialStreamAddresses() bool {
 	return len(p.initialStreamAddresses) > 0
 }
 
+// getSourceTenantID is a bad bad function that returns the source tenant ID only after makePlan is called.
+//
+// TODO(msbutler): remove once spanConfig API is refactored. We need a way to
+// grab the sourceTenantID during distSQL planning. Perhaps we ought to store
+// the sourceTenantID id the job record.
+func (p *replicationFlowPlanner) getSrcTenantID() (roachpb.TenantID, error) {
+	if p.srcTenantID.InternalValue == 0 {
+		return p.srcTenantID, errors.New("make plan must be called before replication flow planner contains srcTenantID")
```

Comment:
Let's make this an AssertionFailedf.

I could see changing the API of this type a little bit so that the usage was more clear.  Namely, rather than having the caller create a `&replicationFlowPlanner{}` and then calling makePlan on it. You could have a constructor that returns a flow planner with the srcTenantID and initial topology set.  Then you could have a method on the struct that was like `generateCandidatePlan()` that would assume it was constructed using the constructor. This function could also assume it was constructed using the constructor.